main.tf

```diff
@@ -57,6 +57,36 @@ resource "aws_iam_service_linked_role" "default" {
   description      = "AWSServiceRoleForAmazonElasticsearchService Service-Linked Role"
 }
 
+# Role that pods can assume for access to elasticsearch and kibana
+resource "aws_iam_role" "elasticsearch_user" {
+  count              = "${var.enabled == "true" ? 1 : 0}"
+  name               = "${module.label.id}-user"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
+  description        = "User of Elasticserach ${module.label.id} cluster"
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  count = "${var.enabled == "true" ? 1 : 0}"
+
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+
+    principals {
+      type        = "AWS"
+      identifiers = ["${compact(concat(var.iam_authorizing_role_arns, var.iam_role_arns))}"]
```

Comment:
why `var.iam_role_arns` here?
this is a list of role ARNs to permit access to the cluster, not to assume the new role

Reply:
The roles listed on line 83 are the ones that are allowed to assume the Elasticsearch User role that is created. There is no point in restricting users who have direct access to Elasticsearch from being able to assume the Elasticsearch user role, since they already have access, but it is preferable for them to assume the role voluntarily to restrict their access when they do not need additional access, so we give them that ability here.